crates/core/src/nats/streams.rs

```diff
@@ -0,0 +1,189 @@
+use std::collections::HashMap;
+
+use futures_util::future::try_join_all;
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+
+use super::{
+    client::NatsClient,
+    types::{
+        JetStreamConfig,
+        NatsConsumer,
+        NatsStorageType,
+        NatsStream,
+        PullConsumerConfig,
+    },
+    NatsError,
+    SubjectName,
+};
+
+#[derive(Debug, EnumIter, Clone, Hash, Eq, PartialEq)]
+pub enum StreamKind {
+    Blocks,
+    Transactions,
+}
+
+impl StreamKind {
+    pub fn get_subjects(&self, prefix: &str) -> Vec<String> {
+        match self {
+            Self::Blocks => vec![SubjectName::Blocks.with_prefix(prefix)],
+            Self::Transactions => vec![
+                SubjectName::Transactions.with_prefix(prefix),
+                SubjectName::TransactionsById.with_prefix(prefix),
+            ],
+        }
+    }
+
+    pub fn get_storage(&self) -> NatsStorageType {
+        match self {
+            Self::Blocks => NatsStorageType::File,
+            Self::Transactions => NatsStorageType::File,
+        }
+    }
```

Comment:
I don't see how all storage won't be `NatsStorageType::File`? To keep it simple, we can avoid this function altogether

Reply:
My idea was that maybe in the future we could have some request-reply, but of course, we can go in a simpler approach